include/deal.II/fe/mapping_q2.h

```diff
@@ -0,0 +1,91 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2020 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_mapping_q2_h
+#define dealii_mapping_q2_h
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/fe/mapping_q_generic.h>
+
+#include <cmath>
+
+DEAL_II_NAMESPACE_OPEN
+
+/*!@addtogroup mapping */
+/*@{*/
+
+
+/**
+ * Implementation of a quadratic mapping from the reference cell to a general
+ * quadrilateral/hexahedron given 9-noded quadrangles or 27-noded hexahedra.
+ *
+ * The mapping implemented by this class is similar to creating a mapping of
+ * the type MappingQ(2) or MappingQGeneric(2). However, the difference is that
+ * this mapping gets the necessary support points from the input file read
+ * with read_msh() instead of computing them or getting them from a manifold.
+ *
+ * @note This class does not support mesh refinement yet. Also, only Gmsh
+ * input files can currently be read and processed to give the necessary
+ * vector of support points.
```

Comment:
Wouldn't it make sense to integrate this into `MappingQCache` - that class already does exactly this kind of data structure?

Reply:
Would you suggest to derive from `MappingQCache` or add a new `MappingQCache::initialize()` function, which takes the additional support poins?

Reply:
I would prefer to simply add a constructor - it is useful for any order of the mapping to be able to provide a set of points. We have the lambda already, but there is no reason against adding another one already holding the array. They can come from a mesh optimization algorithm, for example. One thing we should teach this class - it goes a bit into the direction suggested by @DanielPaukner regarding refinement - is to insert the interpolation facilities between different mesh levels to this class. This way, we would also enable to only fill the active cell positions and interpolate them down to the coarser level for MG adaptivity.

Reply:
Of course I meant to say `initialize()` function rather the constructor.